contents/java/Java.md

```diff
@@ -14,6 +14,275 @@ Authors: John Doe, Jane Doe
+.The event bus uses reflections to observe all registered objects via `register` method for methods annotated with the
+`Subscribe` annotation.
+
+An example implementation (not the actual) of the `Subscribe` annotation might be:
+
+```java
+@Retention(RetentionPolicy.RUNTIME) // Retain this annotation at runtime!
+@Target(ElementType.METHOD) // Only can be applied to methods!
+public @interface Subscribe { }
+```
+
+And that is it! Important parts of the code to note is the first and second line before the declaration. The first line
+tells Java that this annotation must not be discarded during compile time so it will be available during runtime. The
+retention poilcy is there because some annotation do not mean anything after compilation (such as `Override` and
+`SuppressWarnings`), so it does not make sense to keep the annotation after compiling.
+
+The second line just means that this annotation can be applied to methods only.
+
+And the more important part is how the subscriber registry finds all its subscribing methods. The first step is to
+register a class as an event handler and an example of the code is like so:
+
+```java
+
+public class Sheep extends Animal {
+
+    private static final EventsCenter EVENTS_BUS = EventsCenter.getInstance();
+
+    public final String name;
+
+    public Sheep(String nameOfSheep) {
+        name = nameOfSheep;
+        EVENTS_BUS.register(this);
+    }
+    
+}
+```
+
+An example implementation of the `EventsCenter` (with a lot of details left out for simplicity) is like so:
+
+```java
+
+public void register(Class<?> clazz) {
+    findAllEventHandlersInClass(clazz);
+}
+
+public void findAllEventHandlersInClass(Class<?> clazz) {
+    // TypeToken class is provided by Google Guava reflection library
+    Set<? extends Class<?>> supertypes = TypeToken.of(clazz).getTypes().rawTypes();
+
+    for (Class<?> supertype : supertypes) {
+          for (Method method : supertype.getDeclaredMethods()) {
+            if (method.isAnnotationPresent(Subscribe.class) {
+                registerSubscriber(method);
+            }
+        }
+    }
+}
+
+```
+
+The first line of the `findAllEventHandlersInClass` method finds all classes and its parent classes of the registered
+class and converts it to a set. That is if you registered `Sheep extends Animal` as an event handler to the method,
+both `Sheep` and `Animal` will be captured by the first line. The following lines will then examine all their methods
+(during runtime!) for the `Subscribe` annotation and register it so that it will receive the specified event when it 
+is fired. Of course this implementation leaves out a lot of details but you get the idea of how java reflections works.
+
+Another example of reflections might arise when you are trying to get a private field of another class. While this should
```

Comment:
Can consider replacing "might arise when you are trying" with "is" or something else to be consistent with the usage of present tense.